Installing pyinstaller===6.14.2 in FreeBSD is failing with error If I set CC as "clang -pie -fPIE"

Earlier python  3.12.11  and pyinstaller 6.9.0 installed successfully 

But now intallation of pyinstaller 6.14.2  is failing, Now I want to build pyinstaller with CC=clang -pie -fPIE  env variable so that 'run' file of bootlard shared object  which helps my product to be build as DYN file, which is ASLR enabled.


### Context information (for bug reports)

* Output of `pyinstaller --version`: ```(6.14.2)```
* Version of Python: 3.13.5
* Platform:  FreeBSD 
* How you installed Python: python.org/downloads,  and via source code 

## Error durin pip install 
`(virtualEnv) onefs-soujanya-47-1# pip install pyinstaller==6.14.2
Collecting pyinstaller==6.14.2
  Using cached pyinstaller-6.14.2.tar.gz (4.3 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting setuptools>=42.0.0 (from pyinstaller==6.14.2)
  Using cached setuptools-80.9.0-py3-none-any.whl.metadata (6.6 kB)
Collecting altgraph (from pyinstaller==6.14.2)
  Using cached altgraph-0.17.4-py2.py3-none-any.whl.metadata (7.3 kB)
Collecting pyinstaller-hooks-contrib>=2025.5 (from pyinstaller==6.14.2)
  Using cached pyinstaller_hooks_contrib-2025.9-py3-none-any.whl.metadata (16 kB)
Collecting packaging>=22.0 (from pyinstaller==6.14.2)
  Using cached packaging-25.0-py3-none-any.whl.metadata (3.3 kB)
Using cached packaging-25.0-py3-none-any.whl (66 kB)
Using cached pyinstaller_hooks_contrib-2025.9-py3-none-any.whl (444 kB)
Using cached setuptools-80.9.0-py3-none-any.whl (1.2 MB)
Using cached altgraph-0.17.4-py2.py3-none-any.whl (21 kB)
Building wheels for collected packages: pyinstaller
  Building wheel for pyinstaller (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for pyinstaller (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [39 lines of output]
      /tmp/pip-build-env-qzkseeny/overlay/lib/python3.13/site-packages/setuptools/dist.py:759: SetuptoolsDeprecationWarning: License classifiers are deprecated.
      !!

              ********************************************************************************
              Please consider removing the following classifiers in favor of a SPDX license expression:

              License :: OSI Approved :: GNU General Public License v2 (GPLv2)

              See https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#license for details.
              ********************************************************************************

      !!
        self._finalize_license_expression()
      running bdist_wheel
      running clean
      'build/lib' does not exist -- can't clean it
      'build/bdist.freebsd-12.0-CURRENT-amd64' does not exist -- can't clean it
      'build/scripts-3.13' does not exist -- can't clean it
      running build
      running build_bootloader
      No precompiled bootloader found or compile forced. Trying to compile the bootloader for you ...
      Setting top to                           : /tmp/pip-install-hh_pcg6j/pyinstaller_ad4aeb1f32064899a3af5beb5a8152f1/bootloader
      Setting out to                           : /tmp/pip-install-hh_pcg6j/pyinstaller_ad4aeb1f32064899a3af5beb5a8152f1/bootloader/build
      Python Version                           : 3.13.5 (main, Nov 12 2025, 12:52:27) [Clang 6.0.1 (tags/RELEASE_601/final 335540)]
      MSVC target(s)                           : not found
      Checking for 'clang' (C compiler)        : clang -pie -fPIE
      Checking size of pointer                 : 8
      Platform                                 : FreeBSD-64bit detected based on compiler
      Checking for compiler flags -m64         : yes
      Checking for linker flags -m64           : yes
      Checking for header semaphore.h          : not found
      Checking for header sys/sem.h            : not found
      Using semaphore API                      : none
      Checking for library dl                  : not found
      Checking for library pthread             : not found
      Checking for library thr                 : not found
      The configuration failed
      (complete log in /tmp/pip-install-hh_pcg6j/pyinstaller_ad4aeb1f32064899a3af5beb5a8152f1/bootloader/build/config.log)
      ERROR: Failed compiling the bootloader. Please compile manually and rerun
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pyinstaller
Failed to build pyinstaller

[notice] A new release of pip is available: 25.1.1 -> 25.3
[notice] To update, run: pip install --upgrade pip
ERROR: Failed to build installable wheels for some pyproject.toml based projects (pyinstaller)
(virtualEnv) onefs-soujanya-47-1#
`

## Error during manual installation 
`(virtualEnv) onefs-soujanya-47-1# echo $CC
clang -pie -fPIE
(virtualEnv) onefs-soujanya-47-1#
(virtualEnv) onefs-soujanya-47-1#
(virtualEnv) onefs-soujanya-47-1#
(virtualEnv) onefs-soujanya-47-1# python ./waf distclean all -v
'distclean' finished successfully (0.000s)
'all' finished successfully (0.000s)
'distclean' finished successfully (0.000s)
Setting top to                           : /root/soujanya_Regression/pyinstaller-6.14.2/bootloader
Setting out to                           : /root/soujanya_Regression/pyinstaller-6.14.2/bootloader/build
Python Version                           : 3.13.5 (main, Nov 12 2025, 12:52:27) [Clang 6.0.1 (tags/RELEASE_601/final 335540)]
MSVC target(s)                           : not found
Checking for 'clang' (C compiler)        : 20:09:30 runner ['clang', '-pie', '-fPIE', '-dM', '-E', '-']
clang -pie -fPIE
Checking size of pointer                 : 20:09:30 runner ['clang', '-pie', '-fPIE', '-I/var/crash/ese/usr/local/include', '../test.c', '-c', '-o/root/soujanya_Regression/pyinstaller-6.14.2/bootloader/build/.conf_check_872c286f38df153a24ac5bcfe41e9b63/testbuild/test.c.1.o']
20:09:30 runner ['clang', '-pie', '-fPIE', '-I/var/crash/ese/usr/local/include', '../test.c', '-c', '-o/root/soujanya_Regression/pyinstaller-6.14.2/bootloader/build/.conf_check_a8ad87771e9a4b2c98c8c0c285bcd21b/testbuild/test.c.1.o']
20:09:30 runner ['clang', '-pie', '-fPIE', 'test.c.1.o', '-o/root/soujanya_Regression/pyinstaller-6.14.2/bootloader/build/.conf_check_a8ad87771e9a4b2c98c8c0c285bcd21b/testbuild/testprog', '-Wl,-Bstatic', '-Wl,-Bdynamic', '-L/var/crash/ese/usr/local/lib']
8
Platform                                 : FreeBSD-64bit detected based on compiler
Checking for compiler flags -m64         : 20:09:30 runner ['clang', '-pie', '-fPIE', '-I/var/crash/ese/usr/local/include', '-m64', '../test.c', '-c', '-o/root/soujanya_Regression/pyinstaller-6.14.2/bootloader/build/.conf_check_a5af4fed87bfebd5a08c34f51a929f62/testbuild/test.c.1.o']
yes
Checking for linker flags -m64           : 20:09:30 runner ['clang', '-pie', '-fPIE', '-I/var/crash/ese/usr/local/include', '-m64', '../test.c', '-c', '-o/root/soujanya_Regression/pyinstaller-6.14.2/bootloader/build/.conf_check_dba6f373d65fae0d664846477d067c82/testbuild/test.c.1.o']
20:09:30 runner ['clang', '-pie', '-fPIE', '-m64', 'test.c.1.o', '-o/root/soujanya_Regression/pyinstaller-6.14.2/bootloader/build/.conf_check_dba6f373d65fae0d664846477d067c82/testbuild/testprog', '-Wl,-Bstatic', '-Wl,-Bdynamic', '-L/var/crash/ese/usr/local/lib']
yes
Checking for header semaphore.h          : 20:09:30 runner ['clang', '-pie', '-fPIE', '-I/var/crash/ese/usr/local/include', '-m64', '-O2', '-Wall', '-Werror', '-Wno-error=unused-variable', '-Wno-error=unused-function', '-D_REENTRANT', '-D_BSD_SOURCE', '../test.c', '-c', '-o/root/soujanya_Regression/pyinstaller-6.14.2/bootloader/build/.conf_check_0a8784a4fd2a14001f3bce5bc2e9e746/testbuild/test.c.1.o']
not found
Checking for header sys/sem.h            : 20:09:30 runner ['clang', '-pie', '-fPIE', '-I/var/crash/ese/usr/local/include', '-m64', '-O2', '-Wall', '-Werror', '-Wno-error=unused-variable', '-Wno-error=unused-function', '-D_REENTRANT', '-D_BSD_SOURCE', '../test.c', '-c', '-o/root/soujanya_Regression/pyinstaller-6.14.2/bootloader/build/.conf_check_00d7758bd955a8b0ca0e0fb4dc4e9748/testbuild/test.c.1.o']
not found
Using semaphore API                      : none
Checking for library dl                  : 20:09:30 runner ['clang', '-pie', '-fPIE', '-I/var/crash/ese/usr/local/include', '-m64', '-O2', '-Wall', '-Werror', '-Wno-error=unused-variable', '-Wno-error=unused-function', '-D_REENTRANT', '-D_BSD_SOURCE', '../test.c', '-c', '-o/root/soujanya_Regression/pyinstaller-6.14.2/bootloader/build/.conf_check_7f4bda2539af68cac4bfceb0cd41b238/testbuild/test.c.1.o']
not found
Checking for library pthread             : 20:09:30 runner ['clang', '-pie', '-fPIE', '-I/var/crash/ese/usr/local/include', '-m64', '-O2', '-Wall', '-Werror', '-Wno-error=unused-variable', '-Wno-error=unused-function', '-D_REENTRANT', '-D_BSD_SOURCE', '../test.c', '-c', '-o/root/soujanya_Regression/pyinstaller-6.14.2/bootloader/build/.conf_check_c2d12b3fae452eb57d7c6b1216326147/testbuild/test.c.1.o']
not found
Checking for library thr                 : 20:09:30 runner ['clang', '-pie', '-fPIE', '-I/var/crash/ese/usr/local/include', '-m64', '-O2', '-Wall', '-Werror', '-Wno-error=unused-variable', '-Wno-error=unused-function', '-D_REENTRANT', '-D_BSD_SOURCE', '../test.c', '-c', '-o/root/soujanya_Regression/pyinstaller-6.14.2/bootloader/build/.conf_check_8dc66b265fd0939eb0228081eb3f7734/testbuild/test.c.1.o']
not found
The configuration failed
(complete log in /root/soujanya_Regression/pyinstaller-6.14.2/bootloader/build/config.log)
(virtualEnv) onefs-soujanya-47-1#

==========>

<==
[1/2] Compiling ^[[32mbuild/.conf_check_872c286f38df153a24ac5bcfe41e9b63/test.c^[[0m

['clang', '-pie', '-fPIE', '-I/var/crash/ese/usr/local/include', '../test.c', '-c', '-o/root/soujanya_Regression/pyinstaller-6.14.2/bootloader/build/.conf_che
ck_872c286f38df153a24ac5bcfe41e9b63/testbuild/test.c.1.o']
err: clang: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
../test.c:3:35: error: 'test_array' declared as an array with a negative size
            static int test_array[1 - 2 * !(sizeof(void *) == 4)];
                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.

from /root/soujanya_Regression/pyinstaller-6.14.2/bootloader: Test does not build: Traceback (most recent call last):
  File "/root/soujanya_Regression/pyinstaller-6.14.2/bootloader/waflib/Configure.py", line 383, in run_build
    bld.compile()
    ~~~~~~~~~~~^^
  File "/root/soujanya_Regression/pyinstaller-6.14.2/bootloader/waflib/Tools/errcheck.py", line 185, in check_compile
    ret = self.orig_compile()
  File "/root/soujanya_Regression/pyinstaller-6.14.2/bootloader/waflib/Build.py", line 191, in compile
    raise Errors.BuildError(self.producer.error)
waflib.Errors.BuildError: Build failed
 -> task in 'testprog' failed with exit status 1:
        {task 24486782537744: c test.c -> test.c.1.o}
['clang', '-pie', '-fPIE', '-I/var/crash/ese/usr/local/include', '../test.c', '-c', '-o/root/soujanya_Regression/pyinstaller-6.14.2/bootloader/build/.conf_che
ck_872c286f38df153a24ac5bcfe41e9b63/testbuild/test.c.1.o']

from /root/soujanya_Regression/pyinstaller-6.14.2/bootloader: The configuration failed


`

--- a/bootloader/wscript
+++ b/bootloader/wscript
@@ -508,11 +508,18 @@ def configure(ctx):
         # For clang, also enable extra warnings and treat them as errors. Similarly to gcc, do not treat warnings about
         # unused variables and unused functions as errors.
         ctx.env.append_value(
-            'CFLAGS', [
+            'CFLAGS',
+            [
                 '-Wall',
                 '-Werror',
                 '-Wno-error=unused-variable',
                 '-Wno-error=unused-function',
+                # Passing `clang -fPIE -pie` via `CC` environment variable to create position-independent bootloader
+                # executables causes `clang` to emit following warnings when compiling object files due to `-pie`
+                # being linker-stage flag:
+                #   clang: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
+                # Make them non-fatal.
+                '-Wno-error=unused-command-line-argument',
             ]
         )
 

